song-server/src/main/java/org/icgc/dcc/song/server/service/DonorService.java

```diff
@@ -82,18 +84,18 @@ public String update(@NonNull Donor donor) {
     if (donorRepository.update(donor) == 1) {
       return "Updated";
     }
-    return "Failed";
+    return "Failed"; //TODO: [DCC-5644] need to properly handle this. Should an ServerException be thrown?
```

Comment:
Maybe.

We expect to update a record with new data; if nothing has changed, OR if the row we're trying to update doesn't exist, we might in either case get 0 rows modified (depending upon how our back-end database reports things to us).

If the row hasn't changed, we might not want to throw an error; otherwise, we would want to. If want to be really robust, when we get to the transaction handling stage, we could roll back the transaction unless a read gave us back identical data to the update that we tried to do... but that's not a priority for now.

If the database reports back that we've just modified more than one row, then we've got serious issues with our rows not being unique, and we really do have a problem that needs addressing.